client/components/button/test/index.js

```diff
@@ -0,0 +1,1064 @@
+/**
+ * Created by nitrajka on 09/10/16.
+ */
+
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { expect } from 'chai';
+import { shallow } from 'enzyme';
+
+/**
+ * Internal dependencies
+ */
+import Button from '../';
+import Gridicon from 'components/gridicon';
+
+describe( 'Button', function() {
+	describe( 'not compact', function() {
+		describe( 'button tag', function() {
+			const text = 'Text';
+			const icon = ( <Gridicon size={ 18 } icon="arrow-left" /> );
+			const disabled = 'Disabled';
+
+			describe( 'basic styled', function() {
+				it( 'renders with text', function() {
+					const button = shallow( <Button>{ text }</Button> );
+					expect( button ).to.contain( 'Text' );
+					expect( button.is( '.button' ) ).to.equal( true );
+				} );
+
+				it( 'renders with text and icon', function() {
+					const button = shallow( <Button>{ text }<Gridicon size={ 18 } icon="arrow-left" /></Button> );
+					expect( button.find( 'button' ) ).to.contain( 'Text' );
+					expect( button.find( '.button' ) ).to.contain( icon );
+					expect( button.find( 'button' ) ).to.have.className( 'button' );
+					expect( button.find( Gridicon ) ).to.have.prop( 'icon' ).equal( 'arrow-left' );
+				} );
+
+				it( 'renders with icon', function() {
+					const button = shallow( <Button><Gridicon size={ 18 } icon="arrow-left" /></Button> );
+					expect( button.find( 'button' ) ).to.have.className( 'button' );
+					expect( button.find( '.button' ) ).to.contain( icon );
+					expect( button.find( Gridicon ) ).to.have.prop( 'icon' ).equal( 'arrow-left' );
+				} );
+
+				it( 'renders disabled', function() {
+					const button = shallow( <Button disabled>{ disabled }</Button> );
+					expect( button.find( '.button' ) ).to.be.disabled();
```

Comment:
I think calling the `disabled` as a function, even if it is, may not be necessary? At least consistent with how we typically call assertions:

```js
expect( button.find( '.button' ) ).to.be.disabled;
```